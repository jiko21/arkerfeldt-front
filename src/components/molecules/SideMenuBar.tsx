/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react';
import { FC } from 'react';
import { Color } from '@/const/color';
import { MenuItem } from '@/types/MenuItem';

const ulStyle = css`
  background-color: ${Color.BLACK};
  height: 100vh;
  position: sticky;
  width: 240px;
  padding: 210px 0 0 0;
  * {
    list-style-type: none;
  }
`;

const isSelectedStyle = css`
  background-color: ${Color.LIGHT};
`;

const liStyle = css`
  align-items: center;
  color: ${Color.WHITE};
  display: flex;
  font-size: 18px;
  font-weight: 700;
  height: 48px;
  justify-content: center;
  &:hover {
    background-color: ${Color.LIGHT};
  }
`;

type Props = {
  menuItems: MenuItem[];
};

const SideMenuBar: FC<Props> = ({ menuItems }) => (
  <ul css={ulStyle}>
    {menuItems.map((menuItem) => (
      <li
        css={[liStyle, menuItem.isSelected && isSelectedStyle]}
        key={menuItem.url}
      >
        {menuItem.title}
      </li>
    ))}
  </ul>
);

export default SideMenuBar;
