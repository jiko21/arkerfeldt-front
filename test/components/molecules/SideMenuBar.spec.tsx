import SideMenuBar from '@/components/molecules/SideMenuBar';
import { render } from '@testing-library/react';

describe('SideMenuBar.tsx', () => {
  it('should correctly render', () => {
    const { container } = render(
      <SideMenuBar
        menuItems={[
          {
            title: 'Post',
            url: '/post',
            isSelected: true,
          },
          {
            title: 'Layout',
            url: '/layout',
            isSelected: false,
          },
          {
            title: 'User',
            url: '/user',
            isSelected: false,
          },
        ]}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
