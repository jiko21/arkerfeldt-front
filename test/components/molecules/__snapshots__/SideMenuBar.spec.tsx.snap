// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenuBar.tsx should correctly render 1`] = `
.emotion-0 {
  background-color: #2d2d2d;
  height: 100vh;
  position: -webkit-sticky;
  position: sticky;
  width: 240px;
  padding: 210px 0 0 0;
}

.emotion-0 * {
  list-style-type: none;
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 18px;
  font-weight: 700;
  height: 48px;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  background-color: #878686;
}

.emotion-1:hover {
  background-color: #878686;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 18px;
  font-weight: 700;
  height: 48px;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-2:hover {
  background-color: #878686;
}

<div>
  <ul
    class="emotion-0"
  >
    <li
      class="emotion-1"
    >
      Post
    </li>
    <li
      class="emotion-2"
    >
      Layout
    </li>
    <li
      class="emotion-2"
    >
      User
    </li>
  </ul>
</div>
`;
